src/main/java/algorithms/utilities/AlgorithmEstimater.java

```diff
@@ -0,0 +1,13 @@
+package algorithms.utilities;
+
+public class AlgorithmEstimater {
+    /**
+     * The runtime seems to scale with each inputs filesize. The estimate should have at highest ±20% error margin. LEM is the hardest to correctly estimate.
+     * @param sourceSizeKB
+     * @param targetSizeKB
+     * @return
+     */
+    float estimateRuntime(float sourceSizeKB, float targetSizeKB){
+        return 5.0f + sourceSizeKB * targetSizeKB * 3.0f / 2000.0f;
+    }
```

Comment:
The thing I am worried about with this is how the frontend will be notified about this. I agree it's a necessary calculation, but I think it has to happen in the frontend. If it's happening in the backend, we'll have to send the response immediately to say "it will take 32 seconds", and then the frontend will have to reconnect to the backend to get the result. Since this formula isn't so advanced, I recommend moving this logic to the frontend.

Reply:
Thats true. Add it to frontend, once thats done I'll remove it from the backend. (Wait with the pull request)